sopt30th_1st HW/sopt30th_1st HW/LoginViewController.swift

```diff
@@ -0,0 +1,75 @@
+//
+//  LoginViewController.swift
+//  sopt30th_1st HW
+//
+//  Created by 최영린 on 2022/04/06.
+//
+
+import UIKit
+
+class LoginViewController: UIViewController {
+
+    @IBOutlet weak var idTextField: UITextField!
+    @IBOutlet weak var pwTextField: UITextField!
+    @IBOutlet weak var loginButton: UIButton! {
+        didSet{
+            loginButton.isEnabled = false
+        }
+    }
+    @IBOutlet weak var passwordButton: UIButton!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        setBackButton()
+        setLoginButton()
+    }
+    
+    @IBAction func passwordButton(_ sender: Any) {
+        if(pwTextField.isSecureTextEntry){
+            pwTextField.isSecureTextEntry = false
+            passwordButton.setImage(UIImage(named: "PasswordHidden"), for: .normal)
+            }
+        else{
+            pwTextField.isSecureTextEntry = true
+            passwordButton.setImage(UIImage(named: "PasswordShown"), for: .normal)
+        }
+    }
+    
+    @IBAction func loginButtonDidTap(_ sender: Any) {
+        guard let welcomeVC = self.storyboard?.instantiateViewController(withIdentifier: "WelcomeViewController") as? WelcomeViewController else { return }
+        
+        welcomeVC.user = idTextField.text
+        welcomeVC.modalTransitionStyle = .crossDissolve
+        welcomeVC.modalPresentationStyle = .fullScreen
+
+        self.present(welcomeVC, animated: true, completion: nil)
+    }
+    
+    @IBAction func registerButtonDidTap(_ sender: Any) {
+        guard let registerIDVC = self.storyboard?.instantiateViewController(withIdentifier: "RegisterIDViewController") as? RegisterIDViewController else { return }
+        
+        self.navigationController?.pushViewController(registerIDVC, animated: true)
+    }
+    
+    private func setBackButton() {
+        let backBarButtonItem = UIBarButtonItem(title: "", style: .plain, target: nil, action: nil)
+                navigationItem.backBarButtonItem = backBarButtonItem
+    }
+    
+    private func textHandler(_ a: UIAction) -> Void {
+        if self.idTextField.text?.isEmpty == true || self.pwTextField.text?.isEmpty == true{
+            self.loginButton.isEnabled = false
+        } else {
+            self.loginButton.isEnabled = true
+        }
+    }
```

Comment:
(제가 밑에부터 올라와서 보느라,, 똑같은 부분을 또 남기게 되었네요,,)

이 부분은 밑에도 남겨드렸듯이 코드를 줄일 수 있는데요!

idTextField와 pwTextField 둘 중 어느 하나라도 empty라면 버튼은 비활성화 된다라는 말은 곧
== idTextField와 pwTextField는 둘 다 차있어야 버튼이 활성화 될 수 있다

와 같은 맥락으로 해석이 가능할 것 같아요!

```suggestion
    private func textHandler(_ a: UIAction) -> Void {
        self.loginButton.isEnabled = (self.idTextField.text?.isEmpty == false && self.pwTextField.text?.isEmpty == false)
    }
```

그래서 이렇게 해주거나,
혹은,
textField에는, 비어있다는 값을 표현하기에 굳이 !을 붙여주지 않아도 되는 hasText라는 좋은 친구가 있답니다 ,,~

```suggestion
    private func textHandler(_ a: UIAction) -> Void {
        self.loginButton.isEnabled = self.idTextField.hasText && self.pwTextField.hasText
    }
```

이렇게 될 수 있겠네요~!

Reply:
hasText프로퍼티 알아갑니다! 반영할게요:)